qiime2/sdk/result.py

```diff
@@ -108,6 +110,8 @@ def load(cls, filepath):
                    type(result).__name__))
 
         result._archiver = archiver
+        result._read_annotations()
```

Comment:
This isn't the only way to bring up an artifact, so the `_annotations` property won't always be set.

For example `cache.load('my-key')` would skip this line.

Reply:
How about you make `_annotations` a `@property` so that you know that an instance exists, and you can memoize the results, then running `_read_annotations()` in all the correct spots matters way less.

Reply:
Evan and i worked on this offline and have implemented the latter suggestion